### PR TITLE
Let user modify file metadata in the workflow for draft submission

### LINKED ISSUE #26: Add ability to initialize files step in the workflow (1d)
* [ ] Should display list of files already attached to the submission
* [ ] Should highlight flagged files (based on SubmissionActions)
* [ ] Test above

### PR DESCRIPTION
**Pull request by jabrah**
Closes #26 
Partially addresses #21 

I would like some feedback on a couple of things:

* `app/components/file-card/component.js#toggleEditing` function is intended to be an action. When I added the `@action` decorator to this function, things would build, but I would get a runtime error: `TypeError: Cannot set property 'toggleEditing' of undefined`. Calling this function as an action (without the decorator) in the template seems to work as expected, though.
* `app/components/files-list/component.js#filesWithActions` this function breaks if the FilesList component is a Glimmer component. When `@glimmer/component` is imported instead of `@ember/component`, I can't seem to find the named arguments. `this.args` is undefined

### Changes 

* Change the layout of files in the UI (this effects the files display in the Submission Details page too)
  * Now has a table-like layout
  * SubmissionActions are matched to file IDs, if matches are found then the appropriate file is highlighted red and displays the required action description
* In the workflow, file descriptions can be edited

For now, there is no interactive way to "edit" one of the draft submissions that appear in the UI. Instead, you can type in the submission directly. For example, you can go to `http://localhost:4200/submission/new/files?id=5` to load a submission in the workflow

### CODE REVIEW

File: app/components/files-list/component.js [hunk 1]
```diff
@@ -2,46 +2,37 @@ import Component from '@ember/component';
 import { computed, get } from '@ember/object';
 import { A } from '@ember/array';
 
-export default Component.extend({
+export default class FilesList extends Component {
+  // Named properties passed into the component
+  files;
```
Comment (jaredgalanis) [inline app/components/files-list/component.js]: Same issue as file card component.

File: app/components/files-list/component.js [hunk 2]
```diff
@@ -2,46 +2,37 @@ import Component from '@ember/component';
...
   /**
    * Combine files with any associated actions so they can be displayed
+   * Expect 'files' and 'submissionActions' to be proxy objs
+   *
+   * NOTE: Somehow, making this a Glimmer component breaks this function
    */
-  filesWithActions: computed(
-    'files',
-    'submissionActions',
+  @computed(
```
Comment (jaredgalanis) [inline app/components/files-list/component.js]: Again, probably should just get rid fo the computed.

File: app/components/files-list/component.js [hunk 3]
```diff
@@ -2,46 +2,37 @@ import Component from '@ember/component';
...
-          file,
-          hasActions: fileActions.length > 0,
-          actions: fileActions
-        };
-      });
-    }
-  ),
+  )
+  get filesWithActions() {
+    const actions = get(this, 'submissionActions') || A();
```
Comment (jaredgalanis) [inline app/components/files-list/component.js]:
```suggestion
    const actions = this.args.submissionActions || A();
```

File: app/components/files-list/component.js [hunk 4]
```diff
@@ -2,46 +2,37 @@ import Component from '@ember/component';
...
 
-  actions: {
-    /**
-     * Bubble the remove request up to parent component if `removeAction` is defined
-     * @param {File} file file obj
-     */
-    doRemove(file) {
-      this.removeAction(file);
-    },
+    return get(this, 'files').map((file) => {
```
Comment (jaredgalanis) [inline app/components/files-list/component.js]:
```suggestion
    return this.args.files.map((file) => {
```
I tested this manually and if you make this a glimmer component while just accessing the args here it should work (after getting rid of the computed and the property declarations above).

File: app/components/file-card/component.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { computed, set } from '@ember/object';
+import { alias } from '@ember/object/computed';
+
+export default class FileCardComponent extends Component {
+  // Named arguments passed into the component
+  fileGrp;
```
[the remark this answers is not in the document]
Comment (jabrah) [reply app/components/file-card/component.js]: This pattern I repeated a couple times to remind myself what arguments could be passed into the component. JSDoc on the class would be a better place for this info. Will correct in a commit to this PR

File: app/components/file-card/template.hbs [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+            <span class="sr-only">Save</span>
+          </button>
+          <button type="button" class="btn btn-outline-secondary ml-2" {{action this.cancelEdit}}
+              data-test-cancel-edit-btn title="Cancel">
+            <i class="fas fa-times"></i>
+            <span class="sr-only">Cancel</span>
+          </button>
+        {{else}}
+          {{!-- The card handles editing, so no need to pass an editAction --}}
+          <button type="button" class="btn btn-primary" {{action this.toggleEditing}}
```
[the remark this answers is not in the document]
Comment (jabrah) [reply app/components/file-card/template.hbs]: Is the `{{action}}` helper deprecated with Octane, or just different convention?

**Comment by jabrah** [comment]
Think I covered all the suggestions for the two components in question

**Pull request merged**